File: TiposDeDadosES.cpp
#include <iostream>
#include <string>

int main() {
    int integerVar;
    float floatVar;
    double doubleVar;
    char charVar;
    bool boolVar;
    std::string stringVar;

    std::cout << "Digite um valor inteiro: ";
    std::cin >> integerVar;

    std::cout << "Digite um valor decimal (float): ";
    std::cin >> floatVar;

    std::cout << "Digite um valor decimal (double): ";
    std::cin >> doubleVar;

    std::cout << "Digite um caractere: ";
    std::cin >> charVar;

    std::cout << "Digite um valor booleano (0 para falso, 1 para verdadeiro): ";
    std::cin >> boolVar;

    std::cin.ignore(); // Limpar o buffer de entrada antes de ler a string

    std::cout << "Digite uma string: ";
    std::getline(std::cin, stringVar);

    std::cout << "Tipo de Dado\tTamanho (bytes)\tValor\n";
    std::cout << "int\t\t\t" << sizeof(integerVar) << "\t\t" << integerVar << "\n";
    std::cout << "float\t\t\t" << sizeof(floatVar) << "\t\t" << floatVar << "\n";
    std::cout << "double\t\t\t" << sizeof(doubleVar) << "\t\t" << doubleVar << "\n";
    std::cout << "char\t\t\t" << sizeof(charVar) << "\t\t" << charVar << "\n";
    std::cout << "bool\t\t\t" << sizeof(boolVar) << "\t\t" << boolVar << "\n";
    std::cout << "string\t\t\t" << sizeof(stringVar) << "\t\t" << stringVar << "\n";

    return 0;
}

File: TiposDeDadosES.py
import sys

# Declara variáveis para cada tipo de dado
integer_var = None
float_var = None
string_var = None
boolean_var = None

# Solicita ao usuário que insira um valor para cada variável
try:
    integer_var = int(input("Digite um valor inteiro: "))
    float_var = float(input("Digite um valor decimal: "))
    string_var = input("Digite uma string: ")
    boolean_var = input("Digite um valor booleano (True/False): ").lower() == "true"
except ValueError:
    print("Valor inválido. Certifique-se de inserir o tipo de valor correto.")
    sys.exit(1)

# Imprime o nome do tipo, seu tamanho na memória e seu valor atual
print(f"Tipo de Dado\tTamanho (bytes)\tValor")
print(f"int\t\t\t{sys.getsizeof(integer_var)}\t\t{integer_var}")
print(f"float\t\t\t{sys.getsizeof(float_var)}\t\t{float_var}")
print(f"string\t\t\t{sys.getsizeof(string_var)}\t\t{string_var}")
print(f"bool\t\t\t{sys.getsizeof(boolean_var)}\t\t{boolean_var}")

